app/celery/provider_tasks.py

```diff
@@ -127,20 +127,21 @@ def deliver_letter(self, notification_id):
         else:
             current_app.logger.exception(f"RETRY: Letter notification {notification_id} failed")
 
-        self.retry()
+        try:
+            self.retry()
+        except self.MaxRetriesExceededError as e:
+            update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
+            raise NotificationTechnicalFailureException(
+                "RETRY FAILED: Max retries reached. The task deliver_letter failed for notification "
+                f"{notification_id}. Notification has been updated to technical-failure"
+            ) from e
     except DvlaNonRetryableException as e:
         if isinstance(e, DvlaDuplicatePrintRequestException):
             current_app.logger.warning(f"Duplicate deliver_letter task called for notification {notification_id}")
             return
 
         update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
         raise NotificationTechnicalFailureException(f"Error when sending letter notification {notification_id}") from e
-    except self.MaxRetriesExceededError as e:
-        update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
-        raise NotificationTechnicalFailureException(
-            f"RETRY FAILED: Max retries reached. The task deliver_letter failed for notification {notification_id}. "
-            "Notification has been updated to technical-failure"
-        ) from e
     except Exception as e:
         update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
         raise NotificationTechnicalFailureException(f"Error when sending letter notification {notification_id}") from e
```

Comment:
Minor, but I think we could combine these two cases as they seem to be handled the same here?

Reply:
I've updated the code now to combine them